### PR TITLE
Check that getTLS is used from a bound thread.

### PR DESCRIPTION
**Pull request by facundominguez**
This should help while https://github.com/rrnewton/thread-local-storage/pull/10 is not merged.

Another option would be to make a submodule of thread-local-storage, but I'm not sure it is worth the trouble.

### CODE REVIEW

File: jni/src/Foreign/JNI.hs [hunk 1]
```diff
@@ -238,13 +240,23 @@ envTlsRef = unsafePerformIO $ do
       return env; } |]
     newIORef tls
 
+-- | Checks that getTLS is used from a bound thread.
+safeGetTLS :: TLS a -> IO a
+safeGetTLS tls = checkBoundness >> getTLS tls
+  where
+    checkBoundness = do
+      b <- isCurrentThreadBound
+      when (not b) $ do
```
Comment (robinbb) [inline jni/src/Foreign/JNI.hs]: `when (not b)` => `unless b`

File: jni/src/Foreign/JNI.hs [hunk 2]
```diff
@@ -172,6 +173,7 @@ import Foreign.Marshal.Array
 import Foreign.Ptr (Ptr, nullPtr)
 import qualified Language.C.Inline as C
 import qualified Language.C.Inline.Unsafe as CU
+import System.IO
```
Comment (robinbb) [inline jni/src/Foreign/JNI.hs]: LeapYear prefers qualified imports or explicit import lists. Can we do that, here?

File: jni/src/Foreign/JNI.hs (hunk 1 above)
Comment (facundominguez) [inline jni/src/Foreign/JNI.hs]: I have a better time reading `when` that reading `unless`. The explicit negation makes all the difference to me. That said, I will change it to comply with your guidelines ...

Comment (robinbb) [inline jni/src/Foreign/JNI.hs]: We have a hlint job on our non-sparkle LeapYear code which would have complained about this. That's why I brought it up. At some later time, someone will have to fix it...

**Comment by facundominguez** [comment]
Should we close this now that moves are being made to add `thread-local-storage` as a submodule?

**Comment by robinbb** [comment]
Yes, I think we can close.

**Pull request closed without merging**